### PR TITLE
Add PKCS #11 3.0 interface

### PR DESCRIPTION
**Pull request by ZoltanFridrich**
Adds pkcs11v3.0 functions and mechanism into pkcs11.h

### CODE REVIEW

**Comment by ueno** [comment]
Aside from the PR description not matching the actual, I'd suggest starting with only core functions, i.e., without mechanism additions (e.g., `CKM_`, `CKK_`, and `CKD_`, defined in section 6 or later in the spec).

File: common/pkcs11.h [hunk 1]
```diff
@@ -1417,6 +1586,105 @@ struct ck_function_list
...
+  CK_C_GenerateKeyPair C_GenerateKeyPair;
+  CK_C_WrapKey C_WrapKey;
+  CK_C_UnwrapKey C_UnwrapKey;
+  CK_C_DeriveKey C_DeriveKey;
+  CK_C_SeedRandom C_SeedRandom;
+  CK_C_GenerateRandom C_GenerateRandom;
+  CK_C_GetFunctionStatus C_GetFunctionStatus;
+  CK_C_CancelFunction C_CancelFunction;
+  CK_C_WaitForSlotEvent C_WaitForSlotEvent;
+  /* PKCS #11 3.0 functions */
```
Comment (ueno) [inline common/pkcs11.h:1660]: I see a lot of code duplication between `ck_funciton_list` and `ck_function_list_3_0`. Can't we write something like:
```c
#define CK_FUNCTION_LIST_ \
  struct ck_version version; \
  CK_C_Initialize C_Initialize; \
  CK_C_Finalize C_Finalize; \
  CK_C_GetInfo C_GetInfo; \
  ...

struct ck_function_list
{
  CK_FUNCTION_LIST_
};

struct ck_function_list_3_0
{
  CK_FUNCTION_LIST_

  /* PKCS #11 3.0 functions */
  CK_C_GetInterfaceList C_GetInterfaceList;
  CK_C_GetInterface C_GetInterface;
  ...
};

File: common/pkcs11.h [hunk 2]
```diff
@@ -403,6 +412,7 @@ typedef unsigned long ck_key_type_t;
 #define CKK_GOSTR3411		(0x31UL)
 #define CKK_GOST28147		(0x32UL)
 #define CKK_EC_EDWARDS		(0x40UL)
+#define CKK_EC_MONTGOMERY	(0x41UL)
```
Comment (ueno) [inline common/pkcs11.h:415]: Let's do this in a separate commit.

File: common/pkcs11.h [hunk 3]
```diff
@@ -885,7 +896,9 @@ typedef unsigned long ck_mechanism_type_t;
 
 /* From version 3.0 */
 #define CKM_EC_EDWARDS_KEY_PAIR_GEN	(0x1055UL)
+#define CKM_EC_MONTGOMERY_KEY_PAIR_GEN  (0x1056UL)
```
Comment (ueno) [inline common/pkcs11.h:899]: Same, let's skip for now.

File: common/pkcs11.h (hunk 2 above)
Comment (ZoltanFridrich) [reply common/pkcs11.h:415]: but the commit would consist only of 3 lines, is it even worth splitting?

**Review by ueno — APPROVED** [review]
